test/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector.UnitTests/EventLogDataCollectorTests.cs

```diff
@@ -430,39 +423,35 @@ public void WriteEventLogsShouldFilterTestIfMaxValueExceedsEntries()
 /// </summary>
 public class TestableDataCollectionEvents : DataCollectionEvents
 {
-    public override event EventHandler<TestHostLaunchedEventArgs> TestHostLaunched;
-
-    public override event EventHandler<SessionStartEventArgs> SessionStart;
-
-    public override event EventHandler<SessionEndEventArgs> SessionEnd;
-
-    public override event EventHandler<TestCaseStartEventArgs> TestCaseStart;
-
-    public override event EventHandler<TestCaseEndEventArgs> TestCaseEnd;
+    public override event EventHandler<TestHostLaunchedEventArgs>? TestHostLaunched;
+    public override event EventHandler<SessionStartEventArgs>? SessionStart;
+    public override event EventHandler<SessionEndEventArgs>? SessionEnd;
+    public override event EventHandler<TestCaseStartEventArgs>? TestCaseStart;
+    public override event EventHandler<TestCaseEndEventArgs>? TestCaseEnd;
 
     public Delegate[] GetTestHostLaunchedInvocationList()
     {
-        return TestHostLaunched.GetInvocationList();
+        return TestHostLaunched!.GetInvocationList();
```

Comment:
For tests I went ahead with assuming that even when we have a potential null access we dismiss the message (with bang) because the test was working without NRE.

This comment applies generally to the PR.